alicloud/data_source_alicloud_snapshots.go

```diff
@@ -0,0 +1,280 @@
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"creation_time": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"status": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"usage": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceAlicloudSnapshotsRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	args := ecs.CreateDescribeSnapshotsRequest()
+
+	if instanceId, ok := d.GetOk("instance_id"); ok {
+		args.InstanceId = instanceId.(string)
+	}
+	if diskId, ok := d.GetOk("disk_id"); ok {
+		args.DiskId = diskId.(string)
+	}
+	if encrypted, ok := d.GetOk("encrypted"); ok {
+		args.Encrypted = requests.NewBoolean(encrypted.(bool))
+	}
+	if ids, ok := d.GetOk("ids"); ok {
+		args.SnapshotIds = convertListToJsonString(ids.(*schema.Set).List())
+	}
+	if status, ok := d.GetOk("status"); ok {
+		args.Status = status.(string)
+	}
+	if typ, ok := d.GetOk("type"); ok {
+		args.SnapshotType = typ.(string)
+	}
+
+	if diskType, ok := d.GetOk("source_disk_type"); ok {
+		args.SourceDiskType = diskType.(string)
+	}
+	if usage, ok := d.GetOk("usage"); ok {
+		args.Usage = usage.(string)
+	}
+	if dryRun, ok := d.GetOk("dry_run"); ok {
+		args.DryRun = requests.NewBoolean(dryRun.(bool))
+	}
+
+	if v, ok := d.GetOk("tags"); ok {
+		var tags []ecs.DescribeSnapshotsTag
+
+		for key, value := range v.(map[string]interface{}) {
+			tags = append(tags, ecs.DescribeSnapshotsTag{
+				Key:   key,
+				Value: value.(string),
+			})
+		}
+		args.Tag = &tags
+	}
+
+	args.PageSize = requests.NewInteger(PageSizeLarge)
+	args.PageNumber = requests.NewInteger(1)
+	var allSnapshots []ecs.Snapshot
+	for {
+		raw, err := client.WithEcsClient(func(ecsClient *ecs.Client) (interface{}, error) {
+			return ecsClient.DescribeSnapshots(args)
+		})
+		if err != nil {
+			return WrapError(err)
+		}
+		resp := raw.(*ecs.DescribeSnapshotsResponse)
+		allSnapshots = append(allSnapshots, resp.Snapshots.Snapshot...)
+
+		if len(resp.Snapshots.Snapshot) < PageSizeLarge {
+			break
+		}
+
+		if page, err := getNextpageNumber(args.PageNumber); err != nil {
+			return err
+		} else {
+			args.PageNumber = page
+		}
+	}
+
+	var filteredSnapshots []ecs.Snapshot
+	nameRegex, ok := d.GetOk("name_regex")
+	if ok && nameRegex.(string) != "" {
+		var r *regexp.Regexp
+		if nameRegex != "" {
+			r = regexp.MustCompile(nameRegex.(string))
+		}
+		for _, snapshot := range allSnapshots {
+			if r != nil && !r.MatchString(snapshot.SnapshotName) {
+				continue
+			}
+
+			filteredSnapshots = append(filteredSnapshots, snapshot)
+		}
+	} else {
+		filteredSnapshots = allSnapshots
+	}
+
+	if len(filteredSnapshots) < 1 {
+		return fmt.Errorf("Your query returned no results. Please change your search criteria and try again.")
+	}
```

Comment:
Please remove the line 239-241. If there is no any data sources, please return a empty list.